www/docs/content/recipes/digital-products.mdx

```diff
@@ -0,0 +1,1141 @@
+
+```ts title=src/models/product-media.ts
+import { 
+  BeforeInsert, 
+  Column, 
+  Entity,
+  JoinColumn,
+  ManyToOne,
+} from "typeorm"
+import { BaseEntity, ProductVariant } from "@medusajs/medusa"
+import { generateEntityId } from "@medusajs/medusa/dist/utils"
+
+export enum MediaType {
+  MAIN = "main",
+  PREVIEW = "preview"
+}
+
+@Entity()
+export class ProductMedia extends BaseEntity {
+  @Column({ type: "varchar" })
+  name: string
+
+  @Column ({ type: "enum", enum: MediaType, default: "main" })
+  type: MediaType
+
+  @Column({ type: "varchar" })
+  file_key: string
+
+  @Column({ type: "varchar" })
+  variant_id: string
+
+  @ManyToOne(
+    () => ProductVariant,
+    (variant) => variant.product_medias
+  )
+  @JoinColumn({ name: "variant_id" })
+  variant: ProductVariant
+
+  @BeforeInsert()
+  private beforeInsert(): void {
+    this.id = generateEntityId(this.id, "post")
+  }
+}
+```
+
+The entity has the following attributes:
+
+- `name`: a string indicating the file name or a name entered by the merchant.
+- `type`: an enum value indicating the type of file. If the file’s type is `main`, it means that this is the file that customers download when they purchase the product. If its type is `preview`, it means that the file is only used to preview the product variant to the customer.
+- `file_key`: a string that indicates the downloadable file’s key. The key is retrieved by the installed file service, which is covered in the next step, and it’s used later if you want to get a downloadable link or delete the file.
+- `variant_id`: a string indicating the ID of the product variant this file is associated with.
+
+To use this relation on the `ProductVariant` entity as well, extend the entity by creating the file `src/models/product-variant.ts` with the following content:
+
+```ts title=src/models/product-variant.ts
+import { Entity, OneToMany } from "typeorm"
+import { 
+  ProductVariant as MedusaProductVariant,
+} from "@medusajs/medusa"
+import { ProductMedia } from "./product-media"
+
+@Entity()
+export class ProductVariant extends MedusaProductVariant {
+  @OneToMany(() => ProductMedia, (media) => media.variant)
+  product_medias: ProductMedia[]
+}
+```
+
+If you’re using TypeScript, it’s recommended to create the file `src/index.d.ts` with the following content:
+
+<!-- eslint-disable max-len -->
+
+```ts title=src/index.d.ts
+import { ProductMedia } from "./models/product-media"
+
+export declare module "@medusajs/medusa/dist/models/product-variant" {
+  declare interface ProductVariant {
+    product_medias: ProductMedia[];
+  }
+}
+```
+
+This ensures that your IDE’s autocomplete takes into consideration the new `product_medias` relation on the `ProductVariant` entity.
+
+Next, you need to create a repository for the `ProductMedia` entity. To do that, create the file `src/repositories/product-media.ts` with the following content:
+
+```ts title=src/repositories/product-media.ts
```

Comment:
It's technically not necessary to create a repository when you don't have "custom" repository methods. Maybe it would be worth it to just strip this part

Reply:
That's not actually true, considering there's no `productMediaRepository` registered in the container if it's not created. So, you won't be able to use it in services or endpoints by resolving it with dependency injection